core-job/src/main/java/org/apache/kylin/job/execution/AbstractExecutable.java

```diff
@@ -84,6 +88,35 @@ protected void onExecuteStart(ExecutableContext executableContext) {
         getManager().updateJobOutput(getId(), ExecutableState.RUNNING, info, null);
     }
 
+    private void onExecuteFinishedWithRetry(ExecuteResult result, ExecutableContext executableContext) {
+        Throwable exception;
+        int nRetry = 0;
+        do {
+            nRetry++;
+            exception = null;
+            try {
+                onExecuteFinished(result, executableContext);
+            } catch (Exception e) {
+                logger.error(nRetry + "th retries for onExecuteFinished fails due to {}", e);
+                if (isMetaDataPersistException(e)) {
+                    exception = e;
+                    try {
+                        Thread.sleep(1000L * (long) Math.pow(4, nRetry));
```

Comment:
Math.pow(4, nRetry) may be super large, how about sleep a constant period?

Reply:
This logic is for abnormal case. The retry interval becomes larger and larger. There are lots of strategies for retry. I'm open for this.